### ISSUE TITLE
[UI] Allow for developer to set custom values

### ISSUE THREAD

**Issue by richard-cox** [epinio/ui#140]
- See https://github.com/epinio/epinio/issues/1252
- The UI should allow the user to provide custom yaml (form is questions.yaml if available) dependent on the application chart/template when the app is created/edited

**Comment by richard-cox** [comment]
We'll need to get together with the backend team to understand the feature before starting

**Comment by richard-cox** [comment]
As part of this, or straight after, we should ensure https://github.com/epinio/epinio/issues/1792 is possible / has a reasonable UX

**Comment by andreas-kupries** [comment]
To pull in things from the slack disccusion

 - https://github.com/epinio/docs/pull/223
 - https://github.com/epinio/epinio/blob/main/acceptance/helpers/machine/appcharts.go#L52
 -  https://github.com/epinio/epinio/blob/main/acceptance/api/v1/application_validate_cv_test.go#L103
 - https://github.com/epinio/epinio/blob/main/acceptance/apps_test.go#L1371
 - https://github.com/epinio/epinio/blob/main/acceptance/apps_test.go#L1408

> Repo: `epinio/application`
File: `config/crd/bases/application.epinio.io_appcharts.yaml`
Property: `settings`
A map. Keys are the names of the settings supported by the app chart.
Values are maps describing the setting (type mandatory, plus optional restrictions)
Types are `string`, `bool`, `number`, `integer`.The cli collects arbitrary data from the user via option `--chart-value` (short `-v`).
Value syntax same as for `--env`: `name=value`.
The cli __does not know what settings are actually supported__ by the chart. It just collects the data,
as part of app creation/pushing, and then lets the server throw an error on validation
Validation Endpoint: /namespaces/:namespace/applications/:app/validate-cv
It is a separate endpoint to allow for early validation, i.e. immediately after app creation, before
even uploading sources. To avoid doing the expensive actions (uploading, staging, deploy) if the
data is bad.
Note: The endpoint only triggers the validation and returns the result.
The collected settings were passed to the server during app creation.
The relevant field is the `configuration.settings` map.
There is __no endpoint to ask the server for the settings__ a particular chart may support.

**Comment by thehejik** [comment]
**Test Report**

TL;DR: I have verified this feature and it's working as expected. Found just one cosmetic issue with variable names in UI https://github.com/epinio/ui/issues/225

This is how I tested:
* deploy dev epinio from main `v1.7.1-32-gc7e55591`
* patch epinio-ui image: `kubectl set image -n epinio deployment/epinio-ui epinio-ui=ghcr.io/epinio/epinio-ui:latest-next`
* clone helm charts repo `git clone git@github.com:epinio/helm-charts`
* modify the application chart to accept custom user variables and use their values as deployment annotations:
```
diff --git a/chart/application/templates/deployment.yaml b/chart/application/templates/deployment.yaml
index 50fb987..d95da46 100644
--- a/chart/application/templates/deployment.yaml
+++ b/chart/application/templates/deployment.yaml
@@ -15,6 +15,11 @@ spec:
   template:
     metadata:
       annotations:
+        myAnnotationString: {{ .Values.userConfig.myAnnotationString | quote }}
+        myAnnotationEnum: {{ .Values.userConfig.myAnnotationEnum | quote }}
+        myAnnotationInteger: {{ .Values.userConfig.myAnnotationInteger | quote }}
+        myAnnotationNumber: {{ .Values.userConfig.myAnnotationNumber | quote }}
+        myAnnotationBool: {{ .Values.userConfig.myAnnotationBool | quote }}
         app.kubernetes.io/name: {{ .Values.epinio.appName | quote }}
         {{- include "epinio-application.annotations" . | nindent 8 }}
         {{- with .Values.epinio.start }}
```
* pack the application helm chart by `helm package .` and expose the resulting `tgz` on a public webserver
* create a new AppChart resource using custom app variables and pointing to my custom app chart tgz, register to kubernetes/epinio:
```
kubectl apply -f - <<EOF
apiVersion: application.epinio.io/v1
kind: AppChart
metadata:
  namespace: epinio
  name: myannotation
spec:
  shortDescription: Custom epinio deployment myannotation
  description: Epinio deployment with support for custom annotations over .Values.userConfig.myannotation
  helmChart: http://1.2.3.4/epinio-application-0.1.26.tgz
  settings:
    appListeningPort:
      type: 'integer'
    myAnnotationString:
      type: string
    myAnnotationEnum:
      type: string
      enum:
        - "first"
        - "second"
        - "third"
    myAnnotationInteger:
      type: integer
      minimum: '1024'
      maximum: '2048'
    myAnnotationNumber:
      type: number
      minimum: '1'
      maximum: '2'
    myAnnotationBool:
      type: bool
EOF
```
* deploy a new app (any - in this case I used a custom python http server listening on port 1234) using the new AppChart and define values as follows, also verify the input fields are accepting just the values allowed by type and min./max. limits.
![image](https://user-images.githubusercontent.com/12828077/232818035-1fc106ec-3512-4105-8cb0-b3b7e6dbe032.png)
* verify the deployed app has correct values set in its deployment annotations resource:
```
kubectl get pods rpython-server-22a43ae3d879a77acd37c3c63af51159021baf01-5dnkn5k -n workspace -o json | jq '.metadata.annotations'
{
  "myAnnotationBool": "true",
  "myAnnotationEnum": "second",
  "myAnnotationInteger": "1024",
  "myAnnotationNumber": "1.5",
  "myAnnotationString": "mycustomstring"
}
```

**Comment by thehejik** [comment]
Verified fixed in `v1.8.0-rc1` by following the testcase above.
```json
kubectl get pods rpyhttp-custom-bdb737776709789ffd8c2994bf344e4b5e6f5765-6bqn5nn -n workspace -o json | jq '.metadata.annotations'
{
  "app.kubernetes.io/name": "pyhttp-custom",
  "epinio.io/created-by": "admin",
  "myAnnotationBool": "true",
  "myAnnotationEnum": "first",
  "myAnnotationInteger": "1025",
  "myAnnotationNumber": "1.5",
  "myAnnotationString": "blah"
}
```
